src/components/ModelsPage/modelsPage.util.js

```diff
@@ -17,12 +17,50 @@ illegal under applicable law, and the grant of the foregoing license
 under the Apache 2.0 license is conditioned upon your compliance with
 such restriction.
 */
-import React from 'react'
+import React, { useCallback, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import { useYaml } from '../../hooks/yaml.hook'
+
+import artifactsAction from '../../actions/artifacts'
 
 import { MODEL_ENDPOINTS_TAB, MODELS_TAB, REAL_TIME_PIPELINES_TAB } from '../../constants'
 
 export const ModelsPageContext = React.createContext({})
 
+export const ModelsProvider = ({ children }) => {
```

Comment:
rename to `ModelsPageProvider`

Reply:
also, I think it should be in a separate file. For example `ModelsPageProvider.js`